spotify.go

```diff
@@ -0,0 +1,286 @@
+package sync
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"os"
+	"regexp"
+	//"sort"
+	"strings"
+
+	"github.com/zmb3/spotify"
+	"golang.org/x/oauth2"
+)
+
+var (
+	ErrNotReady = fmt.Errorf("client is not ready, make sure user has authenticated the app")
+	ErrNoEnv    = fmt.Errorf("client key and secret is not set in environment variables")
+
+	scopes = []string{spotify.ScopePlaylistModifyPublic, spotify.ScopeUserReadPrivate}
+)
+
+type Spotify struct {
+	Done chan bool
```

Comment:
Instead of making this channel public, return this from a function. Just like how `context` does it.

Reply:
What do you mean by that?

Reply:
```
type Spotify struct {
	done chan bool
}

func (s *Spotify) Done() (chan bool) {
	return s.done
}
```

This way, user will not be able to change the `chan` or close it themselves.